### PR TITLE
added: to express handler

### PR DESCRIPTION
**Pull request by gunzip**
ref. https://github.com/teamdigitale/italia-backend/blob/master/src/app.ts#L70

### CODE REVIEW

**Comment by digitalcitizenship** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  Please include a Pivotal story at the beginning of the PR title (see below).
  </td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>

  Please include a description of your PR changes.
  </td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:book:</td>
      <td>

  :tada: - congrats on your new release
  </td>
    </tr>
  </tbody>
</table>


  Example of PR titles that include pivotal stories:

  * single story: `[#123456] my PR title`
  * multiple stories: `[#123456,#123457,#123458] my PR title`

    
<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: src/express.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+import { IResponse } from "./responses";
+
+/**
+ * Convenience method that transforms a function (handler),
+ * which takes an express.Request as input and returns an IResponse,
+ * into an express controller.
+ */
+export function toExpressHandler<T>(
+  handler: (req: express.Request) => Promise<IResponse<T>>
+): <P>(req: express.Request, res: express.Response, object?: P) => void {
```
Comment (cloudify) [inline src/express.ts]: shouldn't the return type be `express.RequestHandler` ?

File: src/express.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  return async (req, res, object) => {
+    const response = await handler.call(object, req);
```
Comment (cloudify) [inline src/express.ts]: I don't think binding the `handler` to an object should be a concern of this function - if needed, it should be a responsibility of the logic that provides the `handler` to `toExpressHandler`

Comment (gunzip) [reply src/express.ts]: this is intended as drop in replacement to the [referred method](https://github.com/teamdigitale/italia-backend/blob/master/src/app.ts#L70) already used in the italia-backend project. if we drop the object parameter we must do a refactor there cc @lussoluca

File: src/express.ts (hunk 1 above)
Comment (gunzip) [reply src/express.ts]: yes we get rid of the object param, see below

Comment (cloudify) [reply src/express.ts]: good

File: src/express.ts (hunk 2 above)
Comment (cloudify) [reply src/express.ts]: yes let's do that

**Pull request merged**